src/shared/sam/debugger/goSamDebug.ts

```diff
@@ -0,0 +1,185 @@
+    if (!modulePath) {
+        throw new Error(`Cannot find go.mod for: ${filepath}`)
+    }
+
+    return path.dirname(modulePath.fsPath)
+}
+
+/**
+ * Gathers and sets launch-config info by inspecting the workspace and creating
+ * temp files/directories as needed.
+ *
+ * Does NOT execute/invoke SAM, docker, etc.
+ */
+export async function makeGoConfig(config: SamLaunchRequestArgs): Promise<GoDebugConfiguration> {
+    if (!config.baseBuildDir) {
+        throw Error('invalid state: config.baseBuildDir was not set')
+    }
+
+    config.codeRoot =
+        config.codeRoot ??
+        pathutil.normalize(await getSamProjectDirPathForFile(config?.templatePath ?? config.documentUri!!.fsPath))
+
+    if (!config.codeRoot) {
+        throw Error('missing launch.json, template.yaml, and failed to discover project root')
+    }
+
+    let localRoot: string | undefined
+    let remoteRoot: string | undefined
+
+    config.debugPort = config.debugPort ?? -1
+    config.codeRoot = pathutil.normalize(config.codeRoot)
+    config.debuggerPath = GO_DEBUGGER_PATH
+
+    // Always generate a temporary template.yaml, don't use workspace one directly.
+    config.templatePath = await makeInputTemplate(config)
+
+    const isImageLambda = isImageLambdaConfig(config)
+
+    if (isImageLambda && !config.noDebug) {
+        config.containerEnvVars = {
+            _AWS_LAMBDA_GO_DEBUGGING: '1',
+            _AWS_LAMBDA_GO_DELVE_API_VERSION: '2',
+            _AWS_LAMBDA_GO_DELVE_PATH: path.join(GO_DEBUGGER_PATH, 'dlv'),
+            _AWS_LAMBDA_GO_DELVE_LISTEN_PORT: config.debugPort.toString(),
+        }
+    }
+
+    // if provided, use the user's mapping instead
+    if (config.lambda?.pathMappings !== undefined && config.lambda.pathMappings.length > 0) {
+        const mappings = config.lambda.pathMappings
+        if (mappings.length !== 1) {
+            getLogger().warn(
+                'This language only supports a single path mapping entry. Taking the first entry in the list.'
+            )
+        }
+        localRoot = mappings[0].localRoot
+        remoteRoot = mappings[0].remoteRoot
+    }
+
+    //  Make a go launch-config from the generic config.
+    const goLaunchConfig: GoDebugConfiguration = {
+        ...config, // Compose.
+        type: 'go',
+        processName: 'godelve',
+        request: 'attach',
+        mode: config.noDebug ? undefined : 'remote',
+        runtimeFamily: RuntimeFamily.Go,
+        preLaunchTask: undefined,
+        host: 'localhost',
+        port: config.debugPort,
+        skipFiles: [],
+        debugArgs: isImageLambda ? undefined : ['-delveAPI=2'],
+        localRoot: localRoot ?? config.codeRoot,
+        remoteRoot: remoteRoot ?? '/var/task',
+    }
+
+    return goLaunchConfig
+}
+
+/**
+ * @param debuggerPath Installation path for the debugger
+ * @param isWindows Flag for making a windows script
+ * @param forceDirect Sets GOPROXY to direct to prevent DNS failures
```

Comment:
```suggestion
 * @param forceDirect Sets GOPROXY=direct to prevent DNS failures, for use in tests *only*. https://golang.org/ref/mod#module-proxy
```